src/ERC1155A.sol

```diff
@@ -281,85 +281,77 @@ abstract contract ERC1155A is IERC1155A {
     //////////////////////////////////////////////////////////////*/
 
     /// @inheritdoc IERC1155A
-    function registerSERC20(
-        uint256 id,
-        string memory name,
-        string memory symbol
-    )
-        external
-        virtual
-        override
-        returns (address)
-    {
-        if (synthethicTokenId[id] != address(0)) revert SYNTHETIC_ERC20_ALREADY_REGISTERED();
-
-        address syntheticToken = address(new sERC20(name, symbol));
-        synthethicTokenId[id] = syntheticToken;
-
-        return synthethicTokenId[id];
-    }
+    function registerSERC20(uint256 id) external virtual override returns (address);
```

Comment:
I assume  you want to delegate the erc20 creating itself to another contract.
Even then it might still be good to keep the logic with `synthethicTokenId[]` inside this contract, to isolate it an to prevent mistakes.
```solidity
if (synthethicTokenId[id] != address(0)) revert SYNTHETIC_ERC20_ALREADY_REGISTERED();
address syntheticToken = createToken(id); // done elsewhere
synthethicTokenId[id] = syntheticToken;
return synthethicTokenId[id];
```

Reply:
Added